### PR TITLE
updating google-cloud-nio to 0.79.0?

### PR DESCRIPTION
**Pull request by lbergelson**
* updating google-cloud-nio from 0.62.0 -> 0.74.0
* fixes confusing gcloud warning about end user credentials that was output on every start up
* improves https://github.com/broadinstitute/gatk/issues/5447 by reducing the warning to an info

### CODE REVIEW

**Comment by cmnbroad** [comment]
@lbergelsons looks like `RandomDNA.java` needs to have it's import changed to `com.google.common.annotations.VisibleForTesting`

**Comment by lbergelson** [comment]
@cmnbroad 

The warning -> info change is from https://github.com/googleapis/google-auth-library-java/pull/199
That didn't get rid of the stack trace though, just made it an info trace which is still gross. I made a pr which is merged but not yet released which will move the stack trace to debug. 

It should be in 0.75 I think...
https://github.com/googleapis/google-auth-library-java/pull/214

This includes two other prs that I made to fix other output issues in the auth library which made it into the 0.74 release. 
https://github.com/googleapis/google-auth-library-java/pull/205 
https://github.com/googleapis/google-auth-library-java/pull/207

**Comment by lbergelson** [comment]
The only way to test would be scraping the logs, which we could do... but it seems gross.

**Comment by cmnbroad** [comment]
Yeah, probably not worth it.

**Comment by lbergelson** [comment]
There seems to be a change in how gcs recognizes directories that breaks things

Something in `Files.isDirectory` changed.

Previously:
`Files.isDirectory(IOUtils.getPath("gs://hellbender/test/resources/parallel_copy/")) == true`
now it's false.

Weirdly, without the slash it's now true where previously it wasn't...

@jean-philippe-martin This seems related to https://github.com/googleapis/google-cloud-java/pull/3775

**Comment by lbergelson** [comment]
Toggling the psuedodirs option doesn't seem to fix the problem...

**Comment by jean-philippe-martin** [comment]
That certainly looks like a bug! I'm off for a few days but will take a look when I return.

**Comment by jean-philippe-martin** [comment]
The situation is rather odd: you seem to have **both** a file **and** a folder named "`gs://hellbender/test/resources/parallel_copy/`". Observe the first line of output:

```bash
$ gsutil ls gs://hellbender/test/resources/parallel_copy/
gs://hellbender/test/resources/parallel_copy/
gs://hellbender/test/resources/parallel_copy/bar.txt
gs://hellbender/test/resources/parallel_copy/foo.txt
```

The file exists and is empty. For evidence, observe that I can cat the file (note how it behaves differently from normal folders):

```bash
$ gsutil cat gs://hellbender/test/resources/
CommandException: No URLs matched: gs://hellbender/test/resources/
$ gsutil cat gs://hellbender/test/resources/parallel_copy/
$
```

So NIO is technically correct, `gs://hellbender/test/resources/parallel_copy/` is a file. Did you do anything unusual with regards to this path?

**Comment by jean-philippe-martin** [comment]
The issue linked above indicates a way you could have ended in this twilight zone: apparently this is the default behavior of the GCS web UI and its "create folder" button.

**Comment by droazen** [comment]
@lbergelson Should this PR be closed? It seems like we'll need a PR to bump us to a different gcloud release that fixes the issue you ran into here.

**Comment by lbergelson** [comment]
I've bumped this to 0.78.0 which includes a fix for the directory issue, it's not JP's fix so we'll see if it works as expected...

**Comment by lbergelson** [comment]
Bleh... the newest version is broken in a worse way.  They fixed a bug where some service providers weren't being shaded correctly, but it broke the shadow jar by accidentally shading the service provider for the actual FileSystemProvider.

**Comment by lbergelson** [comment]
I opened a new issue to track this new bug here: https://github.com/googleapis/google-cloud-java/issues/4403

Hopefully this is an easy fix.

**Comment by droazen** [comment]
A fix for the shading issue has been merged into google-cloud-java, but is not yet released.

**Comment by ldgauthier** [comment]
Does that mean 4.1 is still going to use the NIO from your personal repo?

**Comment by droazen** [comment]
@ldgauthier No, we've been on the official google-cloud-java release for a long time now -- our fork is long dead. We're just trying to update to the latest version to fix some edge-case issues with empty files in GCS buckets, but it's not a blocker for the 4.1 release.

File: src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java [hunk 1]
```diff
@@ -383,6 +383,7 @@ public static CloudStorageConfiguration getCloudStorageConfiguration(int maxReop
             // enable requester pays and indicate who pays
             builder = builder.autoDetectRequesterPays(true).userProject(requesterProject);
         }
+        builder.usePseudoDirectories(true);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java]: Can you add a comment explaining what this does?

**Pull request merged**